### PR TITLE
Read metadata location from json file rather than yml

### PR DESCRIPTION
**Pull request by JHjava**
- The application uses the json file but the ruby script currently reads
from yml. Make the ruby script ready from json too so the metadata files are accurate should/when we use the dropwizard application in the future. 
- The #{environment} environment variable is already set in jenkins job builder https://github.com/alphagov/verify-jenkins-job-builder/blob/master/jobs/ci-1/eidas-metadata-aggregator.yml, so it can find the correct file for each env. 
- All metadata we use at the moment is public and therefore there is no
need to supply the tls cert. Simplify the script by remove the reading of the tls from the ruby script.
- Remove the now redundant yml files containing the country metadata and redundant tls cert for croatia. 
- Will disable prod job before merging so I can confirm it runs successfully in the other envs.

### CODE REVIEW

**Comment by tomrosier** [comment]
Are theses logs suppose to be in the commit? `log/debug/metadata-aggregator.log.2019-01-08.gz`

**Comment by JHjava** [comment]
@tomrosier No they're not. Thanks for pointing out.

File: upload_to_s3.rb [hunk 1]
```diff
@@ -1,27 +1,24 @@
...
-metadata_list = YAML.load_file File.join(__dir__, metadata_list_file)
-abort("Error: File supplied is not valid YAML") unless metadata_list
 
-def metadata_content url, tls_cert
-  puts "USING TLS CERT: #{tls_cert}" if tls_cert
-  optional_tls_cert = "--cacert #{tls_cert}" if tls_cert
-  `curl #{optional_tls_cert} -A "Mozilla" --max-time 15 #{url}`
-end
+metadata_json_list = File.read(metadataLocation)
+countryUrlList = JSON.parse(metadata_json_list)
```
Comment (tomrosier) [inline upload_to_s3.rb]: Just make the variable names consistant as we are mixing camel and under_scorie

File: upload_to_s3.rb [hunk 2]
```diff
@@ -35,11 +32,12 @@ def aws_put object, content
 s3 = Aws::S3::Resource.new(region: aws_region)
 updated_metadata = []
 
-metadata_list.each do |country, metadata|
-  url = metadata['url']
-  s3_object_key = url.unpack('H*')[0].downcase
-  updated_metadata << s3_object_key
-  aws_put s3.bucket(bucket_name).object(s3_object_key), metadata_content(url, metadata['tls_cert'])
+countryList.each do |country|
```
Comment (tomrosier) [inline upload_to_s3.rb]: you might be able todo, `|country, url|` save extracting it later.

File: upload_to_s3.rb [hunk 3]
```diff
@@ -35,11 +32,12 @@ def aws_put object, content
...
+    puts "Pushing #{country[0]} to S3 bucket"
+    url = country[1]
+    s3_object_key = url.unpack('H*')[0].downcase
+    updated_metadata << s3_object_key
```
Comment (tomrosier) [inline upload_to_s3.rb]: what does this line do? is the `<<` doing bit shifting?

File: src/main/resources/integration/MetadataSourceConfiguration.json [hunk 4]
```diff
@@ -1,10 +1,13 @@
 {
   "metadataUrls": {
-    "Stub Country metadata": "https://idp-stub-joint.ida.digital.cabinet-office.gov.uk/stub-country/ServiceMetadata",
-    "Poland metadata": "https://plnode.eidas.gov.pl/EidasNode/ServiceMetadata",
-    "Italy metadata": "https://service.eid.gov.it/ServiceMetadata",
+    "Stub Country metadata": "https://ida-stub-idp-integration.cloudapps.digital/stub-country/ServiceMetadata",
```
[the remark this answers is not in the document]
Comment (JHjava) [reply src/main/resources/integration/MetadataSourceConfiguration.json]: Done

File: upload_to_s3.rb (hunk 2 above)
Comment (JHjava) [reply upload_to_s3.rb]: Good suggestion. Done.

File: upload_to_s3.rb (hunk 1 above)
Comment (JHjava) [reply upload_to_s3.rb]: Yep good point. Done.

File: upload_to_s3.rb (hunk 3 above)
Comment (JHjava) [reply upload_to_s3.rb]: This line was already there. Having a quick google, I believe it puts the s3_object_key to the end of the updated_metadata array so it's in order.

File: upload_to_s3.rb [hunk 5]
```diff
@@ -1,27 +1,24 @@
...
-require 'yaml'
 require 'cgi'
+require 'json'
 
 aws_region = ENV['AWS_REGION']
 bucket_name = ENV['BUCKET_NAME']
-metadata_list_file = ENV['METADATA_FILE']
 
-abort('Please supply AWS_REGION, BUCKET_NAME, METADATA_FILE as environment variables') unless aws_region && bucket_name && metadata_list_file
+abort('Please supply AWS_REGION, BUCKET_NAME, ENVIRONMENT as environment variables') unless aws_region && bucket_name && environment
```
Comment (tomrosier) [inline upload_to_s3.rb]: If i'm being picky it be nice to drop the abort bit onto the next line and wrap it, as its a fairly long line as is.
```
unless aws_region && bucket_name && environment
   abort('Please supply AWS_REGION, BUCKET_NAME, ENVIRONMENT as environment variables')
end 
```

**Pull request merged**